pkg/systemd/quadlet/quadlet.go

```diff
@@ -1865,13 +1869,26 @@ func addNetworks(quadletUnitFile *parser.UnitFile, groupName string, serviceUnit
 				// the systemd unit name is $serviceName.service
 				networkServiceName := networkUnitInfo.ServiceFileName()
 
-				serviceUnitFile.Add(UnitGroup, "Requires", networkServiceName)
+				if isNetworkUnit {
+					serviceUnitFile.Add(UnitGroup, "Requires", networkServiceName)
+				} else {
+					// do not add a strong dependency;
+					// instead, fail if that container is not started.
+					serviceUnitFile.Add(UnitGroup, "Requisite", networkServiceName)
```

Comment:
What is the use case for this? Why would someone want this to fail vs just starting the unit?

Reply:
Creating and starting a container is non-trivial and has side-effect; it is unlike creating a network.

Instead of pulling in a strong dependency on another container and implicitly starting it, I think a "Requisite" can avoid mis-configuation; the user needs to explicitly enable (`systemctl enable`) that container.

From a semantic perspective, if a container reuses another container's network, the former is usually a part of the latter. (For example, the latter is a server, and the former is a database.) The user usually enables both of them, not only the former.

I've also mentioned this behavior in the docs.

Reply:
I disagree.

The same can be said when referencing a `.image` or `.build` file. In order for this unit to start successfully, the unit for the other `.container` must start as well so I don't see a reason forcing the user into manually starting two services.

Please keep in mind that since Quadlet units are considered transient, they cannot be enabled by systemctl.

Using a different behavior is just an GitHub issue waiting to be opened (even when it is explained in the man page)

Reply:
Ok then. Modified.